### PR TITLE
feat(MCDA): 18479 Add new MCDA transformation options (log, log_epsilon, cube_root), fix square_root transformation

### PR DESCRIPTION
**Pull request by albaranau**
https://kontur.fibery.io/Tasks/Task/Add-new-MCDA-transformation-options-18479

## Summary by CodeRabbit

- **New Features**
  - Introduced new mathematical transformations including cube root, log one, and log epsilon.
  - Enhanced support for axis transformation data in the UI.

- **Bug Fixes**
  - Updated existing transformation functions to correctly handle negative values.

- **Improvements**
  - Enhanced localization for mathematical transformations.
  - Improved calculation logic for transformations to incorporate dataset minimum values for better accuracy.

- **Dependency Updates**
  - Added `@reatom/async` and `@reatom/hooks` packages for asynchronous functionality and hooks support.

- **Refactor**
  - Renamed `availableBivariateAxisesAtom` to `availableBivariateAxesAtom` for consistency.

- **Tests**
  - Added new tests for the introduced mathematical transformations.

### CODE REVIEW

File: src/core/localization/translations/en/common.json [hunk 1]
```diff
@@ -120,8 +120,14 @@
       "reset_limits_to_default": "Reset limits to default",
       "exclude": "Exclude",
       "as_values_on_limits": "As values on limits",
-      "natural_logarithm": "Logarithm",
-      "square_root": "Square root",
+      "transformations": {
+        "no_transformation": "No transformation",
+        "natural_logarithm": "Logarithm",
```
Comment (Komzpa) [inline src/core/localization/translations/en/common.json]: So is this a natural log or log10? Natural has e as base.

Comment (albaranau) [reply src/core/localization/translations/en/common.json]: That is a natural log, it's for an old transformation option (I have a question for tomorrow discussion about whether we need to keep it).
New ones use Log10

**Review by Komzpa — APPROVED** [review]

File: src/core/logical_layers/renderers/stylesConfigs/mcda/types.ts [hunk 2]
```diff
@@ -45,7 +46,12 @@ export interface MCDAConfig {
 }
 
 export type Normalization = 'max-min' | 'no';
-export type TransformationFunction = 'no' | 'natural_logarithm' | 'square_root';
+export type TransformationFunction =
```
Comment (Natallia-Harshunova) [inline src/core/logical_layers/renderers/stylesConfigs/mcda/types.ts:49]: I'd personally prefer to use enum here not to hard code values, smth like that
```
export enum TransformationFunction {
  No = 'no',
  SquareRoot = 'square_root',
  CubeRoot = 'cube_root',
  Log = 'log',
  LogEpsilon = 'log_epsilon',
}
```
and use it `TransformationFunction.No` and so on

File: src/core/logical_layers/renderers/stylesConfigs/mcda/calculations/index.ts [hunk 3]
```diff
@@ -6,11 +6,20 @@ import type { MCDAConfig, TransformationFunction } from '../types';
 const equalSentiments = (a: Array<string>, b: Array<string>) =>
   a.length === b.length && a.every((x, i) => x === b[i]);
 
+// x - hex value
+// min, max - current range min and max
+// datasetMin - minimal value in dataset for indicator, constant value
```
Comment (Natallia-Harshunova) [inline src/core/logical_layers/renderers/stylesConfigs/mcda/calculations/index.ts]: This comment related to `transform `function, maybe place it before it, not whole interface

File: src/core/logical_layers/renderers/stylesConfigs/mcda/calculations/index.ts [hunk 4]
```diff
@@ -58,17 +69,45 @@ class Calculations<T> implements IsomorphCalculations<T> {
       /* square_root: (sqrt(x) - sqrt(min)) / (sqrt(max) - sqrt(min)) */
       case 'square_root':
         return {
-          tX: this.math.sqrt(x),
-          tMin: this.math.sqrt(min),
-          tMax: this.math.sqrt(max),
+          tX: this.math.mult(this.math.sign(x), this.math.sqrt(this.math.abs(x))),
```
Comment (Natallia-Harshunova) [inline src/core/logical_layers/renderers/stylesConfigs/mcda/calculations/index.ts:75]: Can you please update comment of this transformation, it got me confused

**Review by Natallia-Harshunova — APPROVED** [review]

File: src/core/logical_layers/renderers/stylesConfigs/mcda/types.ts (hunk 2 above)
Comment (albaranau) [reply src/core/logical_layers/renderers/stylesConfigs/mcda/types.ts:49]: We're not really using enums in this project currently, but I understand the idea. Think that's something we should discuss with FE team if we want to move to using enums

File: src/core/logical_layers/renderers/stylesConfigs/mcda/calculations/index.ts (hunk 3 above)
Comment (albaranau) [reply src/core/logical_layers/renderers/stylesConfigs/mcda/calculations/index.ts]: Moved the comment, also changed it a bit to make more informative

File: src/core/logical_layers/renderers/stylesConfigs/mcda/calculations/index.ts (hunk 4 above)
Comment (albaranau) [reply src/core/logical_layers/renderers/stylesConfigs/mcda/calculations/index.ts:75]: Done! Good point

**Pull request merged**